docs/input.md

```diff
@@ -90,20 +90,199 @@ def setup(): # Only drawn once, so just using setup()
+
+#### key
+
+A string that is the last key pressed
+
+##### Notes
+- This variable **does not clear**, meaning if a key is pressed it will remain the value until a **new key** is pressed
+- This variable will be the uppercase letter if ++shift++ + a letter is pressed i.e. ++shift++ + ++a++ would be `#!python key == "A"`
+- This variable will be a symbol if ++shift++ + a number is pressed i.e. ++shift++ + ++3++ would be `#!python key == "$"`
+
+
+##### Special Keys
+
+| Key           | Value                          |
+|---------------|--------------------------------|
+| ++up++        | `#!python key == "ArrowUp"`    |
+| ++down++      | `#!python key == "ArrowDown"`  |
+| ++left++      | `#!python key == "ArrowLeft"`  |
+| ++right++     | `#!python key == "ArrowRight"` |
+| ++tab++       | `#!python key == "Tab"`        |
+| ++ctrl++      | `#!python key == "Control"`    |
+| ++alt++       | `#!python key == "Alt"`        |
+| ++esc++       | `#!python key == "Escape"`     |
+| ++win++       | `#!python key == "Meta"`       |
+| ++backspace++ | `#!python key == "Backspace"` |
+
+**Example(s):**
+
+*Print out each key as they are pressed (I pressed **just** the ++a++ key in this example)*
+
+
+```python hl_lines="8"
+%%ignite
+def setup():
+    size(500,500)
+
+def draw():
+    background(255)
+    text_size(32)
+    text(key, 250, 200)
+```
+
+Results in:
+![key example](img/key.png){: loading=lazy }
+
+#### keys_held()
+
+```python
+keys_held(key)
+```
+
+**Parameters**
+
+- key: (str) The key you want to check for
+
+**Returns**
+
+bool; Returns True if the provided key is held, else False.
+
+
+
+*Print `#!python "b key held"` if the ++b++ key is held*
+
+```python hl_lines="7"
+%%ignite
+
+def setup():
+    size(200, 200)
+
+def draw():
+    if keys_held("b"):
+        print("b key held")
+```
+
+Results in:
+
+![keys_held() example](img/keys_held.png){: loading=lazy }
+
+#### key_pressed()
+
+```python
+def key_pressed():
+    # Your code goes here
+```
+
+This is a user definable function that activates when **any** key is held.
+
+Example(s):
+
+*Print a key if it's pressed*
+
+```python hl_lines="9 10"
+%%ignite
+
+def setup():
+    size(200, 200)
+
+def draw():
+    ... # Does nothing
+
+def key_pressed():
+    print(key)
+```
+
+Results in:
+
+![key_pressed() example](img/key_pressed.png){: loading=lazy }
+
+#### key_released()
+
+```python
+def key_released():
+    # Your code goes here
+```
+
+This is a user definable function that activates when **any** key is released.
+
+Example(s):
+
+*Print a key if it's pressed*
+
+```python hl_lines="9 10"
+%%ignite
+
+def setup():
+    size(200, 200)
+
+def draw():
+    ... # Does nothing
+
+def key_released():
+    print(key)
+```
+
+Results in:
+
+![key_released() example](img/key_released.png){: loading=lazy }
+
+#### key_repeated()
+
+```python
+def key_repeated():
+    # Your code goes here
+```
+
+This is a user definable function that activates when **any** key is held.
```

Comment:
May want to point out that this activates repeatedly when any key is held?

Reply:
Fair point, I will make an adjustment

Reply:
Adjustment made in latest commit to branch